heat/core/manipulations.py

```diff
@@ -1331,6 +1426,182 @@ def squeeze(x, axis=None):
+        if shapes.count(shapes[0]) != num_arrays:
+            raise ValueError(
+                "All DNDarrays in sequence must have the same shape, got shapes {}".format(shapes)
+            )
+        splits = list(array.split for array in arrays)
+        if splits.count(splits[0]) != num_arrays:
+            raise ValueError(
+                "All DNDarrays in sequence must have the same split axis, got splits {}"
+                "Check out the heat.resplit() documentation.".format(splits)
+            )
+        devices = list(array.device for array in arrays)
+        if devices.count(devices[0]) != num_arrays:
+            raise RuntimeError(
+                "DNDarrays in sequence must reside on the same device, got devices {}".format(
+                    devices
+                )
+            )
+        balance = list(array.is_balanced() for array in arrays)
+        if balance.count(balance[0]) != num_arrays:
+            raise RuntimeError(
+                "DNDarrays distribution must be balanced across ranks, is_balanced() returns {}"
+                "You can balance a DNDarray with the balance_() method.".format(balance)
+            )
+    else:
+        array_shape, array_split, array_device = arrays_metadata[0][:3]
+        # extract torch tensors
+        t_arrays = list(array._DNDarray__array for array in arrays)
+        # output dtype
+        t_dtypes = list(t_array.dtype for t_array in t_arrays)
+        t_array_dtype = t_dtypes[0]
+        if t_dtypes.count(t_dtypes[0]) != num_arrays:
+            for d in range(1, len(t_dtypes)):
+                t_array_dtype = (
+                    t_array_dtype
+                    if t_array_dtype is t_dtypes[d]
+                    else torch.promote_types(t_array_dtype, t_dtypes[d])
+                )
+            t_arrays = list(t_array.type(t_array_dtype) for t_array in t_arrays)
+        array_dtype = types.canonical_heat_type(t_array_dtype)
+
+    # sanitate axis
+    axis = stride_tricks.sanitize_axis(array_shape + (num_arrays,), axis)
+
+    # output shape and split
+    stacked_shape = array_shape[:axis] + (num_arrays,) + array_shape[axis:]
+    if array_split is not None:
+        stacked_split = array_split + 1 if axis <= array_split else array_split
+    else:
+        stacked_split = None
+
+    # sanitate output
+    if out is not None:
+        if not isinstance(out, dndarray.DNDarray):
+            raise TypeError("expected out to be None or ht.DNDarray, but was {}".format(type(out)))
+        if out.dtype is not array_dtype:
+            raise TypeError("expected out to be {}, but was {}".format(array_dtype, out.dtype))
+        if out.gshape != stacked_shape:
+            raise ValueError(
+                "expected out.shape to be {}, got {}".format(out.gshape, stacked_shape)
+            )
+        if out.split is not stacked_split:
+            raise ValueError("expected out.split to be {}, got {}".format(out.split, stacked_split))
+    # end of sanitation
```

Comment:
unnecessary, covered by moving out of the code block (or the sanitize output comment can be moved into the block)

Reply:
> unnecessary, covered by moving out of the code block (or the sanitize output comment can be moved into the block)

You're right, I'm just leaving notes for my future self, the one that's about to implement the sanitation module. It will be gone soon enough. Thanks.

Reply:
are you talking about making a sanitation module specifically for the stack module? because your tests are pretty specific, like they are all required to be the exact same, correct?

Reply:
i thought that this was in a different section. sorry. too many things on my mind today i guess

Reply:
> are you talking about making a sanitation module specifically for the stack module? because your tests are pretty specific, like they are all required to be the exact same, correct?

No, sorry, I'm talking about a general sanitation/validation module, see #468